netty/src/test/java/io/grpc/netty/NettyClientStreamTest.java

```diff
@@ -407,8 +408,13 @@ public void getRequestSentThroughHeader() {
     ArgumentCaptor<CreateStreamCommand> cmdCap = ArgumentCaptor.forClass(CreateStreamCommand.class);
     verify(writeQueue).enqueue(cmdCap.capture(), eq(true));
     assertThat(ImmutableListMultimap.copyOf(cmdCap.getValue().headers()))
-        .containsEntry(AsciiString.of(":path"), AsciiString.of(
-            "//testService/test?" + BaseEncoding.base64().encode(msg)));
+        .containsEntry(
+            AsciiString.of(":path"),
+            AsciiString.of(
+                "//testService/test?"
+                    + GRPC_PAYLOAD_BIN_KEY
+                    + "="
+                    + BaseEncoding.base64().encode(msg)));
```

Comment:
Add an assert that the base64-encoded `msg` does not include `=`. Ditto for OkHttp

Reply:
Picking this back up after ~7 months to clear PR backlog...

This was a good catch: the base64 encoder uses `=` for padding. The call here should actually be to [`base64Url()`](https://google.github.io/guava/releases/16.0/api/docs/com/google/common/io/BaseEncoding.html#base64Url()), to use `-` and `_` instead of `+` and `/`. grpc/grpc has an equivalent to `base64Url`, but it also uses `=` for padding. AFAICT using the `=` reserved character will still "work" but violates [RFC 1738](https://www.ietf.org/rfc/rfc1738.txt). We should just URL encode the padding character here, but this requires decoding capability on the server side. Will update the grpc/grpc PR and then update this again (sometime this week, hopefully; in <7 months, worst case ;))

Reply:
Virtually all base64 versions use `=` for padding. The difference is whether they use padding at all. For Guava, we just need to call `omitPadding()`. I don't think we should UrlEncode the padding character, but that's a bit of a spec-level decision.